backend/service/aws/aws.go

```diff
@@ -80,29 +87,88 @@ func New(cfg *anypb.Any, logger *zap.Logger, scope tally.Scope) (service.Service
 			return nil, err
 		}
 
-		c.clients[region] = &regionalClient{
-			region: region,
-			dynamodbCfg: &awsv1.DynamodbConfig{
-				ScalingLimits: &awsv1.ScalingLimits{
-					MaxReadCapacityUnits:  ds.MaxReadCapacityUnits,
-					MaxWriteCapacityUnits: ds.MaxWriteCapacityUnits,
-					MaxScaleFactor:        ds.MaxScaleFactor,
-					EnableOverride:        ds.EnableOverride,
-				},
-			},
-
-			s3:          s3.NewFromConfig(regionCfg),
-			kinesis:     kinesis.NewFromConfig(regionCfg),
-			ec2:         ec2.NewFromConfig(regionCfg),
-			autoscaling: autoscaling.NewFromConfig(regionCfg),
-			dynamodb:    dynamodb.NewFromConfig(regionCfg),
-			sts:         sts.NewFromConfig(regionCfg),
+		if _, ok := c.accounts[c.currentAccountAlias]; !ok {
+			c.accounts[c.currentAccountAlias] = &accountClients{
+				alias:   accountAlias,
+				regions: ac.Regions,
+				clients: map[string]*regionalClient{},
+			}
 		}
+
+		c.createRegionalClients(c.currentAccountAlias, region, ds, regionCfg)
+	}
+
+	if err := c.configureAdditonalAccountClient(ac.AdditionalAccounts, ds, awsHTTPClient); err != nil {
+		return nil, err
 	}
 
 	return c, nil
 }
 
+func (c *client) configureAdditonalAccountClient(accounts []*awsv1.AWSAccount, ds *awsv1.ScalingLimits, awsHTTPClient *http.Client) error {
+	for _, account := range accounts {
+		accountRoleARN := fmt.Sprintf("arn:aws:iam::%s:role/%s", account.AccountNumber, account.IamRole)
+		// For doing STS calls it does not matter which region client we are using, as they are not bounded by region
+		// we choose just the first region client
+		stsClient := c.accounts[c.currentAccountAlias].clients[c.accounts[c.currentAccountAlias].regions[0]].sts
+		assumeRoleProvider := stscreds.NewAssumeRoleProvider(stsClient, accountRoleARN)
+		credsCache := aws.NewCredentialsCache(assumeRoleProvider)
+
+		for _, region := range account.Regions {
+			regionCfg, err := config.LoadDefaultConfig(context.TODO(),
```

Comment:
any thoughts on how to reduce the duplication of config creation code on line :76?

Reply:
Yeah played around with it a bit and this should reduce some of the duplication and make it easier to handle common options, let me know what you think.